core/src/main/java/org/apache/calcite/plan/RelOptMaterializations.java

```diff
@@ -60,7 +60,8 @@
    *         materialized views used in the transformation.
    */
   public static List<Pair<RelNode, List<RelOptMaterialization>>> useMaterializedViews(
-      final RelNode rel, List<RelOptMaterialization> materializations) {
+      final RelNode rel, List<RelOptMaterialization> materializations,
+      List<SubstitutionVisitor.UnifyRule> materializationRules) {
     final List<RelOptMaterialization> applicableMaterializations =
         getApplicableMaterializations(rel, materializations);
```

Comment:
Oops, this is a breaking change.

Reply:
Maybe we can be compatible with the original interface.In my project, this is how to register external materialized view matching rules.

Reply:
If we have to make this change, let us deprecate it first.

Reply:
ok, update it.

Reply:
Should we keep the default implementation without custom materialized recognition rules? Method overloading.

Reply:
@chunweilei Can we be compatible with existing interfaces?